Source/WebCore/rendering/RenderTextControl.cpp

```diff
@@ -138,12 +139,18 @@ void RenderTextControl::hitInnerTextElement(HitTestResult& result, const LayoutP
 float RenderTextControl::getAverageCharWidth()
 {
     float width;
-    if (style().fontCascade().fastAverageCharWidthIfAvailable(width))
-        return width;
+    const FontCascade& font = style().fontCascade();
+    if (font.fastAverageCharWidthIfAvailable(width)) {
+        // We apply roundf() only if the fractional part of 'width' is >= 0.5
+        // because:
+        // * We have done it for a long time.
+        // * Removing roundf() would make the intrinsic width smaller, and it
+        //   would have a compatibility risk.
+        return std::max(width, roundf(width));
+    }
```

Comment:
This function should return LayoutUnit and  use LayoutUnit's ceiling. It would make it consistent with the rest of the intrinsic size ceiling you find across the codebase.